File: Tests/CSharp/Checksums/Checksums.cs
// **********************************************************************
//
// Copyright (c) 2009-2015 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Test
{
    public class Checksums
    {
        public static String checksum()
        {
            return Ice.SliceChecksums.checksums.FirstOrDefault(i => i.Key.Equals("::Test::Person")).Value;
        }
    }
}

File: Tests/Cpp/Checksums/Checksums.cpp
// **********************************************************************
//
// Copyright (c) 2009-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#include <Ice\Ice.h>
#include <Test.h>

#include "Checksums.h"

bool Test::checksums()
{
	Ice::CommunicatorPtr communicator = Ice::initialize();
	Ice::SliceChecksumDict checksums = Ice::sliceChecksums();
    return checksums.find("::Test::Person") != checksums.end();
}